### DISCUSSION TITLE
How to run dcm4chee-arc-ui2 locally?

### DISCUSSION THREAD

**Discussion by lqkalla** [dcm4che/dcm4chee-arc-light#3625] [Q&A]
Hi!

I would like to run dcm4chee-arc-ui2 locally to develop a feature as part of my thesis, but having issues. Could you help?

I am getting the following error on "ng serve" without any changes to the codebase:
```
Schema validation failed with the following errors:
  Data path "" should NOT have additional properties(crossOrigin).

```

What are the recommended Angular, Node and npm versions? Should installing be done via yarn instead of npm?

The builder under serve in angular.json file gives a warning:
<img width="561" alt="image" src="https://user-images.githubusercontent.com/38937521/160486910-fce5f611-5bfb-48e7-9f36-a5b0d155057a.png">

This is the sequence I followed to try to get the service running, am I doing something wrong?
```
yarn install
ng build
ng serve
```

Thank you kindly!

**Comment by shral** [comment]
Hi,
We don't use `ng serve`, because it is a java project, you only build the angular project (navigate to `dcm4chee-arc-light/dcm4chee-arc-ui2`, and run `ng build --watch` for developer mode or `ng build --aot --prod` for production mode) and then run (in another terminal) `mvn install -D secure=all` that will create a war file in the target folder of the UI project, you have to deploy that in the wildfly (if you are using a dockerized version in linux, you probably have an folder in `/var/local/dcm4chee-arc/ui/deployments/`, you could simply copy the war file inthere).
